projects/core/src/main/resources/data/computercraft/lua/bios.lua

```diff
@@ -728,24 +742,24 @@ if fs.exists(".settings") then
 end
 
 -- Run the shell
+local shellOk
 local ok, err = pcall(parallel.waitForAny,
     function()
-        local sShell
-        if term.isColour() and settings.get("bios.use_multishell") then
-            sShell = "rom/programs/advanced/multishell.lua"
-        else
-            sShell = "rom/programs/shell.lua"
+        shellOk = os.run({}, settings.get("bios.shell_path"))
+        if shellOk then
```

Comment:
Is there a reason this part of logic is changed and extra variable is checked?

Reply:
yes, when the shell errors os.run emits false, and does not error. this was never caught before and so after 1 second of the error displaying, the computer shuts down. The only reason we see the error in the first place is because we're afforded that 1 second by the shutdown rom program saying goodbye.

Reply:
This actually runs a little deeper, in that we don't know on what terminal the error from os.run was rendered on. Ideally we'd redirect to the native one, like we do when the whole parallel loop errors. I almost wish os.run gave the error string like pcall does so we could just `printError` it like we do with the parallel loop.

Reply:
latest commit extracts os.run logic so we're able to print the error on the correct terminal